### ISSUE TITLE
hildingr - Delevering can be blocked by other positions

### ISSUE THREAD

**Issue by sherlock-admin** [sherlock-audit/2023-05-Index-judging#315]
hildingr

medium

# Delevering can be blocked by other positions

## Summary

AAVE3 introduces new conditions on when collateral can be removed. One new condition is that collateral withdrawal is blocked if any other collateral token's LTV has been set to 0.

This introduces a scenario where a AaveLeverageExtension can be blocked from delevering. This was not an issue with AAVE2 since this constraint did not exist.

## Vulnerability Detail

When a collateral's LTV is set to 0 on AAVE3 all other collateral are blocked from withdrawal until that token is completely removed.

A separate component in a SetToken can therefore block the Aave3LeverageStrategyExtension from withdrawing collateral.

It i always necessary to withdraw collateral to delever in the Aave3LeverageStrategyExtension since the `delever()` function in Aave3LeverageModule always withdraws collateral to repay debt. 

If we can not withdraw collateral we are blocked from delevering until the other component has fully been withdrawn. 

## Impact

Not being able to withdraw collateral puts the leverage position at significant risk of liquidation since the safety functionality to delever through the `AaveLeverageStrategyExtension#ripcord()` and  `AaveLeverageStrategyExtension#disengage()` is blocked.

## Code Snippet

https://github.com/sherlock-audit/2023-05-Index/blob/main/index-protocol/contracts/protocol/modules/v1/AaveV3LeverageModule.sol#L313-L338

## Tool used

Manual Review

## Recommendation

A  solution that guarantees that delevering can always happen requires a separate safety feature similar to the ripcord that does not rely on withdrawing collateral.

This could be done by giving anybody the ability to delever by directly transferring tokens to repay the debt and receiving payment in terms of minted SetTokens in proportion to how much was repaid + a bonus fee. The NAV for users would be intact other than a fee. This would only be callable in rare occasions, similar to he ripcord.

I also recommend that a note is added to the documentation that states that SetTokens with multiple positions with AAVE3 collateral introduce a new risk for leverage positions.

Duplicate of https://github.com/sherlock-audit/2023-05-Index-judging/issues/159
